build failure in function convert_array_std140

### Godot version

4.1 stable

### System information

openSUSE Leap 15.4 and 15.5

### Issue description

Building godot 4.1 for openSUSE in
https://build.opensuse.org/package/show/home:cunix:godot/godot
fails for Leap with

```
g++ -o drivers/gles3/storage/material_storage.linuxbsd.editor.x86_64.o -c -std=gnu++17 -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -pipe -pthread -O2 -Wall -Wshadow -Wno-misleading-indentation -Wno-strict-overflow -Wno-type-limits -isystem thirdparty/glad -DTOOLS_ENABLED -DDEBUG_ENABLED -DNDEBUG -DNO_EDITOR_SPLASH -DTOUCH_ENABLED -DFONTCONFIG_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -D_REENTRANT -DPULSEAUDIO_ENABLED -DDBUS_ENABLED -DSPEECHD_ENABLED -DXKB_ENABLED -DJOYDEV_ENABLED -DUDEV_ENABLED -DLINUXBSD_ENABLED -DUNIX_ENABLED -D_FILE_OFFSET_BITS=64 -DX11_ENABLED -DVULKAN_ENABLED -DGLES3_ENABLED -DMINIZIP_ENABLED -DBROTLI_ENABLED -DZSTD_STATIC_LINKING_ONLY -DUSE_VOLK -DVK_USE_PLATFORM_XLIB_KHR -DGLAD_ENABLED -DGLES_OVER_GL -DGLAD_GLX_NO_X11 -DHAVE_MNTENT -Ithirdparty/freetype/include -Ithirdparty/libpng -Ithirdparty/volk -Ithirdparty/vulkan -Ithirdparty/vulkan/include -Ithirdparty/zstd -Ithirdparty/zlib -Ithirdparty/brotli/include -Iplatform/linuxbsd -I/usr/include/miniupnpc -I. -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/speech-dispatcher -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/libxkbcommon drivers/gles3/storage/material_storage.cpp

[...]

 In file included from drivers/gles3/storage/material_storage.cpp:42:0:
 ./servers/rendering/storage/variant_converters.h: In function 'Vector<P> convert_array_std140(const Variant&, bool)':
 ./servers/rendering/storage/variant_converters.h:281:45: error: 'const class Variant' has no member named 'operator T'; did you mean 'operator Vector<Color>'?
      convert_item_std140<T, P>(item.operator T(), offset, true);
                                              ^
                                              operator 104
[...]

 scons: *** [drivers/gles3/storage/material_storage.linuxbsd.editor.x86_64.o] Error 1
 scons: building terminated because of errors.
```

build log is currently here:
https://build.opensuse.org/public/build/home:cunix:godot/15.4/x86_64/godot/_log

Building previous stable godot version 4.0.3 succeeds:
https://build.opensuse.org/public/build/games/15.4/x86_64/godot/_log

Build of 4.1 for Tumbleweed works:
https://build.opensuse.org/public/build/home:cunix:godot/openSUSE_Tumbleweed/x86_64/godot/_log

I'm not sure what is happening,
This might be introduced by
https://github.com/godotengine/godot/pull/74937
and only visible with older build tools.

For example Tumbleweed uses gcc-c++ 13, while Leap has version 7.5.

Perhaps this is too old nowadays and unsupported?


### Steps to reproduce

branch
https://build.opensuse.org/package/show/games/godot
on obs, update and build godot 4.1


### Minimal reproduction project

pre project selection

--- a/SConstruct
+++ b/SConstruct
@@ -621,22 +621,12 @@ if methods.using_gcc(env):
             "Couldn't detect compiler version, skipping version checks. "
             "Build may fail if the compiler doesn't support C++17 fully."
         )
-    # GCC 8 before 8.4 has a regression in the support of guaranteed copy elision
-    # which causes a build failure: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86521
-    elif cc_version_major == 8 and cc_version_minor < 4:
+    elif cc_version_major < 9:
         print_error(
-            "Detected GCC 8 version < 8.4, which is not supported due to a "
-            "regression in its C++17 guaranteed copy elision support. Use a "
-            'newer GCC version, or Clang 6 or later by passing "use_llvm=yes" '
-            "to the SCons command line."
-        )
-        Exit(255)
-    elif cc_version_major < 7:
-        print_error(
-            "Detected GCC version older than 7, which does not fully support "
-            "C++17. Supported versions are GCC 7, 9 and later. Use a newer GCC "
-            'version, or Clang 6 or later by passing "use_llvm=yes" to the '
-            "SCons command line."
+            "Detected GCC version older than 9, which does not fully support "
+            "C++17, or has bugs when compiling Godot. Supported versions are 9 "
+            "and later. Use a newer GCC version, or Clang 6 or later by passing "
+            '"use_llvm=yes" to the SCons command line.'
         )
         Exit(255)
     elif cc_version_metadata1 == "win32":

